pg_metadata/dock.py

```diff
@@ -66,12 +77,81 @@ def __init__(self, parent=None):
         menu.addAction(self.auto_open_dock_action)
         self.config.setMenu(menu)
 
+        # Setting PDF/HTML menu
+        self.save_button.setAutoRaise(True)
+        self.save_button.setToolTip(tr("Save metadata"))
+        self.save_button.setPopupMode(QToolButton.InstantPopup)
+        self.save_button.setIcon(QIcon(QgsApplication.iconPath('mActionFileSave.svg')))
+
+        self.save_as_pdf = QAction(
+            tr('Save content as PDF file'),
+            iface.mainWindow())
+        self.save_as_pdf.triggered.connect(partial(self.export_dock_content, 'pdf'))
+
+        self.save_as_html = QAction(
+            tr('Save content as HTML file'),
+            iface.mainWindow())
+        self.save_as_html.triggered.connect(partial(self.export_dock_content, 'html'))
+
+        menu_save = QMenu()
+        menu_save.addAction(self.save_as_pdf)
+        menu_save.addAction(self.save_as_html)
+        self.save_button.setMenu(menu_save)
+
         self.metadata = QgsProviderRegistry.instance().providerMetadata('postgres')
 
         self.default_html_content()
 
         iface.layerTreeView().currentLayerChanged.connect(self.layer_changed)
 
+    def export_dock_content(self, output_format):
+        if not iface.activeLayer():
+            iface.messageBar().pushWarning(
+                "Select Layer",
+                "You need to select layer to export in PDF or HTML"
+            )
+            return
+        my_settings = self.settings
+        my_settings.setValue("UI/lastFileNameWidgetDir", 'home/user/')
```

Comment:
I don't understand why do you set a value here ? You are hardcoding a value whaterver the user already got in this variable, no ?

Reply:
If the user opened another directory with other tools during previous use of qgis, it will be redirected to them and not / home under linux or Documents for windows